### DISCUSSION TITLE
RecordListOptions and the fields property

### DISCUSSION THREAD

**Discussion by mez** [pocketbase/pocketbase#3643] [Q&A] [answered]
Wondering if this is supported or if I made a mistake. I have two collections `customers` and `invoices`. Invoices has a relation to customers called `customer`. I found the `fields` property and I wanted to use it to select columns from the relations table. Is it possible? Currently only returns the customer `id`

The following code: 
```
const queryOptions: RecordListOptions = {
      fields: 'id, amount, date, status, customer.email, customer.name',
      expand: 'customer',
      filter: pb.filter("customer.name ~ {:query} || customer.email ~ {:query}", {query})
    };

const invoices = await pb.collection('invoices').getList(currentPage, ITEMS_PER_PAGE, queryOptions);
```

Produces the following output:

> [
>   {
>     amount: 1000,
>     customer: 'pixeercqodpdeo3',
>     date: '2022-06-05 00:00:00.000Z',
>     id: '4l0sls6gdga3ydd',
>     status: 'pending'
>   },
> ]

What does the sql translate to when you use the field property combined with the expand property?

**Comment by ganigeorgiev** [comment, marked as answer]
`fields` applies transformation on top of the json response right before returning it to the client, meaning that it follows the record object structure. In your case that would be `..., expand.customer.email, expand.customer.name`.

**Reply by mez** [reply]
Understood. Thank you!
